### DISCUSSION TITLE
Is there any issue on beforeSave() ?

### DISCUSSION THREAD

**Discussion by ariesdg** [filamentphp/filament#12182] [Help] [answered]
### Package

Panel builder

### Package Version

V3.2

### How can we help you?

I am working on EditUser page of a UserResource. I want to have condition before it finally save to database. The condition will halt the saving process if satisfied. However during my testing, the record keeps on saving, so I tried call the halt() directy. And it still saves.

Btw. I am using Spatie Laravel roles and permission package.

This is my EditUser page
```
 protected function beforeSave(): void
 {   
      $this->halt();     
 }
```

This is my component form in UserResource

```
Select::make('roles')
  ->preload()
  ->relationship('roles','name')
  ->columnSpanFull()
  ->multiple()
```

**Comment by leandrocfe** [comment, marked as answer]
You could use a custom rule in the `Select`, because the relationship is saved automatically

```php
->rules([
    function () {
        return function (string $attribute, $value, Closure $fail) {
            if ($value === 'something') {
                $fail('The :attribute is invalid.');
            }
        };
    },
])
```

**Reply by ariesdg** [reply]
Thanks
